test/micro/org/openjdk/bench/jdk/incubator/vector/EqualsIgnoreCaseBenchmark.java

```diff
@@ -0,0 +1,152 @@
+ *  version 2 for more details (a copy is included in the LICENSE file that
+ *  accompanied this code).
+ *
+ *  You should have received a copy of the GNU General Public License version
+ *  2 along with this work; if not, write to the Free Software Foundation,
+ *  Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ *  Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ *  or visit www.oracle.com if you need additional information or have any
+ *  questions.
+ *
+ */
+
+package org.openjdk.bench.jdk.incubator.vector;
+
+import jdk.incubator.vector.ByteVector;
+import jdk.incubator.vector.VectorMask;
+import jdk.incubator.vector.VectorSpecies;
+import org.openjdk.jmh.Main;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Exploration of vectorized latin1 equalsIgnoreCase taking advantage of the fact
+ * that ASCII and Latin1 were designed to optimize case-twiddling operations.
+ */
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Benchmark)
+@Warmup(iterations = 5, time = 1)
+@Measurement(iterations = 5, time = 1)
+@Fork(value = 3, jvmArgsPrepend = {"--add-modules=jdk.incubator.vector"})
+public class EqualsIgnoreCaseBenchmark {
+    static final VectorSpecies<Byte> SPECIES = ByteVector.SPECIES_PREFERRED;
+    private byte[] a;
+    private byte[] b;
+    private int len;
+    @Param({"16", "32", "64", "128", "1024"})
+    private int size;
+
+    @Setup
+    public void setup() {
+        a = ("a\u00e5".repeat(size/2) + "A").getBytes(StandardCharsets.ISO_8859_1);
+        b = ("A\u00c5".repeat(size/2) + "B").getBytes(StandardCharsets.ISO_8859_1);
+        len = a.length;
+    }
+    @Benchmark
+    public void scalar(Blackhole blackhole) {
+        blackhole.consume(scalarEqualsIgnoreCase(a, b, len));
+    }
+
+    @Benchmark
+    public void vectorized(Blackhole blackhole) {
+        blackhole.consume(vectorizedEqualsIgnoreCase(a, b, len));
+    }
+
+    private boolean vectorizedEqualsIgnoreCase(byte[] a, byte[] b, int len) {
+        int i = 0;
+        for (; i < SPECIES.loopBound(b.length); i += SPECIES.length()) {
+            ByteVector va = ByteVector.fromArray(SPECIES, a, i);
+            ByteVector vb = ByteVector.fromArray(SPECIES, b, i);
+            VectorMask<Byte> equal = va.eq(vb);
+
+            // If all bytes are equal, we can skip ahead early
+            if (equal.allTrue()) {
+                continue;
+            }
+
+            // ASCII and Latin-1 were designed to optimize case-twiddling operations
+            ByteVector lowerA = va.or((byte) 0x20);
+
+            // Determine which bytes represent ASCII or Latin-1 letters:
+            VectorMask<Byte> asciiLetter = lowerA.lt((byte) '{').and(lowerA.lt((byte) 0x60).not());
```

Comment:
We do have GT/GE/NE etc comparison operators supported in Vector API, which you can use here and other places in this benchmark. e.g.
You could do lowerA.compare(GE, (byte)0x60) instead of using lt() followed not(). BTW did you mean to use GT here?

You will need to do the following import:
import static  jdk.incubator.vector.VectorOperators.*;

Reply:
Thanks, that's a lot more readable! This checks that bytes are in the range => (byte) 'A', <= (byte) 'A', so I think this "not less than" was a poor man's compare(GE, (byte) 'A') here.

I've updated the code to use LE, GE, NE where appropriate